runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/config/HttpBindingConfig.java

```diff
@@ -252,173 +261,142 @@ public HttpRequestType resolveRequest(
-                if (isValid.get())
+                if (valid.value)
                 {
-                    Validator validator = request.headers.get(header.name());
+                    Validator validator = requestType.headers.get(header.name());
                     if (validator != null)
                     {
                         String16FW value = header.value();
                         if (!validator.read(value.value(), value.offset(), value.length()))
                         {
-                            isValid.set(false);
+                            valid.value = false;
                         }
                     }
                 }
             });
         }
-        return isValid.get();
+        return valid.value;
     }
 
     private boolean validatePathParams(
-        HttpRequestType request)
+        HttpRequestType requestType,
+        String path)
     {
-        boolean isValid = true;
-        for (String name : request.pathParamValues.keySet())
+        Matcher matcher = requestType.pathMatcher.reset(path);
+        boolean matches = matcher.matches();
+        assert matches;
+
+        boolean valid = true;
+        for (String name : requestType.pathParams.keySet())
         {
-            Validator validator = request.pathParams.get(name);
-            if (validator != null)
+            String value = matcher.group(name);
+            if (value != null)
             {
-                String8FW value = request.pathParamValues.get(name);
-                if (!validator.read(value.value(), value.offset(), value.length()))
+                String8FW value0 = new String8FW(value);
+                Validator validator = requestType.pathParams.get(name);
+                if (!validator.read(value0.value(), value0.offset(), value0.length()))
                 {
-                    isValid = false;
+                    valid = false;
                     break;
                 }
             }
         }
-        return isValid;
+        return valid;
     }
 
     private boolean validateQueryParams(
-        HttpRequestType request)
+        HttpRequestType requestType,
+        String path)
     {
-        boolean isValid = true;
-        for (String name : request.queryParamValues.keySet())
+        Matcher matcher = requestType.queryMatcher.reset(path);
+        boolean valid = true;
+        boolean found = matcher.find();
+        while (found)
         {
-            Validator validator = request.queryParams.get(name);
+            String name = matcher.group(1);
+            Validator validator = requestType.queryParams.get(name);
             if (validator != null)
             {
-                String8FW value = request.queryParamValues.get(name);
+                String8FW value = new String8FW(matcher.group(2));
                 if (!validator.read(value.value(), value.offset(), value.length()))
                 {
-                    isValid = false;
+                    valid = false;
```

Comment:
```
value &= validator.read(value.value(), value.offset(), value.length());
```

Reply:
IntelliJ pointed out that `valid` is always false here so it can be simplified to
`valid = validator.read(value.value(), value.offset(), value.length());`